foundations/src/security/mod.rs

```diff
@@ -388,6 +476,31 @@ pub fn enable_syscall_sandboxing(
     Ok(())
 }
 
+/// Gets the secure computing mode of the current thread.
+///
+/// Uses the [prctl(PR_GET_SECCOMP)] syscall, so calling this without an allow_list such as
+/// the following may violate sandboxing rules if called after [`enable_syscall_sandboxing`].
+///
+/// ```
+/// use foundations::security::{ArgCmp, allow_list, common_syscall_allow_lists::RUST_BASICS};
+///
+/// allow_list! {
+///    pub static MY_ALLOW_LIST = [
+///        ..RUST_BASICS
+///    ]
+/// }
+/// ```
+///
+/// [prctl(PR_GET_SECCOMP)]: https://linuxman7.org/linux/man-pages/man2/PR_GET_SECCOMP.2const.html
+pub fn get_current_thread_seccomp_mode() -> BootstrapResult<SeccompMode> {
```

Comment:
Does this need to be `pub`?

Reply:
Doesn't need to be, no. But I figured since we had it, that it could be useful to users of foundations - though I'm not really sure how.

I guess this comes down to "expose it because we have it" vs "keep it private to minimize public surface area unless definitely needed". I lean towards the former in this case, but don't feel strongly either way, so if you'd prefer it be made private I'd be happy to do so.

Reply:
I usually only expose things when we already have use cases in mind for it

Reply:
Yeah that's fair. I've removed `pub` from its definition and moved it down. I left the docstring since it was already written even though it's not "required" any more